### DISCUSSION TITLE
FastApi Loading Issue

### DISCUSSION THREAD

**Discussion by sindhubha** [fastapi/fastapi#10644] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
engine = create_engine("mysql+pymysql://root@localhost:3307/mydatabase",poolclass=NullPool)
SessionLocal = sessionmaker(autocommit=False, autoflush=False, bind=engine)

def get_db():
   
    try:    
        db = SessionLocal()
        yield db
    finally:
        db.close()

@app.post("/demo/")
async def demo(cnx: Session = Depends(get_db)):
    query = (f'...')
    data = cnx.execute(query).fetchall()
```

### Description
API's are not running  while one API is loading and the entire application get's hanging.

### Operating System

Windows

### FastAPI Version

0.95.1

### Pydantic Version

1.10.7

### Python Version

3.19.12

**Comment by JavierSanchezCastro** [comment, marked as answer]
Did you try async connections with sqlalchemy?

Look at this template: https://github.com/Lolomgrofl/fastapi-genesis/blob/master/%7B%7Bcookiecutter.repo_name%7D%7D/app/db.py
